app/src/main/java/explorer/city/com/cityexplorer/view/adapter/DetailsAdapter.kt

```diff
@@ -0,0 +1,35 @@
+package explorer.city.com.cityexplorer.view.adapter
+
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import explorer.city.com.cityexplorer.R
+import explorer.city.com.cityexplorer.model.CityDetail
+import explorer.city.com.cityexplorer.view.adapter.viewHolder.DetailsViewHolder
+
+class DetailsAdapter: RecyclerView.Adapter<DetailsViewHolder>() {
+
+    private val details by lazy { mutableListOf<CityDetail>() }
+
+    fun addItems(items: List<CityDetail>?) {
```

Comment:
This method is badly named. It's called `addItems` but it actually clears the list. Either call it `setItems` or `replaceItems`. There's a big difference.